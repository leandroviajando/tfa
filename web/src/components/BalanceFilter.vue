<template>
  <v-autocomplete
    id="balance-filter"
    :value="selectedBalanceId"
    :label="$t('balance')"
    :items="balances"
    dense
    @change="(value) => $emit('set-selected-balance-id', value)"
  />
</template>

<script>
import { mapActions, mapGetters } from 'vuex';

export default {
  props: {
    selectedBalanceId: {
      type: Number,
      default: null,
    },
  },
  data() {
    return {};
  },
  computed: {
    ...mapGetters(['balances']),
  },
  async created() {
    await this.fetchBalances();
  },
  methods: {
    ...mapActions(['fetchBalances']),
  },
};
</script>
