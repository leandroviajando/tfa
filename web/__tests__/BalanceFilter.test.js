import { createLocalVue, mount } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Vue from 'vue';
import Vuex from 'vuex';
import i18n from '@/plugins/i18n';
import BalanceFilter from '@/components/BalanceFilter';
import services_axios from '@/services/axios';
import { flushPromises } from '../utils';

describe('receiving partner filter component test', () => {
  const localVue = createLocalVue();
  Vue.use(Vuetify);
  localVue.use(Vuex);
  let vuetify;
  let getters;
  let state;
  let actions;

  const testBalances = [
    { value: null, text: 'All' },
    { value: 12, text: 'Default USD - Dahabshiil' },
    { value: 13, text: 'Commission - USD - Dahabshiil' },
    { value: 14, text: 'Fees Western Union - MTN Uganda' },
  ];

  const instantiateState = () => {
    // Set initial state
    state = {
      balances: testBalances,
    };
    getters = {
      balances: () => state.balances,
    };
    actions = {
      fetchBalances: jest.fn(),
    };
    return new Vuex.Store({
      state,
      getters,
      actions,
    });
  };

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const emitMock = jest.fn();

  const getWrapper = (selectedBalanceId = null) => {
    const store = instantiateState();
    const wrapper = mount(BalanceFilter, {
      propsData: {
        selectedBalanceId,
      },
      localVue,
      vuetify,
      store,
      i18n,
    });

    wrapper.vm.$emit = emitMock;

    return wrapper;
  };

  it('renders as expected, using balances from state', async () => {
    services_axios.get = jest.fn(() => Promise.resolve({ data: testBalances }));
    const wrapper = await getWrapper();
    await flushPromises();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('sets selected balance if provided', async () => {
    services_axios.get = jest.fn(() => Promise.resolve({ data: testBalances }));
    const wrapper = await getWrapper(12);
    await flushPromises();
    const filter = wrapper.find('#balance-filter');
    expect(filter.element.value).toBe('Default USD - Dahabshiil');
  });
});
