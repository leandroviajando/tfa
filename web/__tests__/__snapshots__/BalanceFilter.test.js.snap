// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`receiving partner filter component test renders as expected, using balances from state 1`] = `
<div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-select v-autocomplete">
  <div class="v-input__control">
    <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-5" class="v-input__slot">
      <div class="v-select__slot"><label for="balance-filter" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Balance</label><input id="balance-filter" type="text" autocomplete="off">
        <div class="v-input__append-inner">
          <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
        </div><input type="hidden">
      </div>
      <div class="v-menu">
        <!---->
      </div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
